waspc/src/Wasp/Psl/Ast/ConfigBlock.hs

```diff
@@ -0,0 +1,54 @@
+module Wasp.Psl.Ast.ConfigBlock
+  ( ConfigBlock (..),
+    ConfigBlockType (..),
+    ConfigBlockKeyValue (..),
+    configBlockName,
+    configBlockKeyValues,
+  )
+where
+
+type Name = String
+
+-- | Represents a config block in the PSL.
+--   For example, in the following PSL:
+--   ```
+--   generator client {
+--     provider = "prisma-client-js"
+--   }
+--   ```
+--   The config block would be `ConfigBlock Generator "client" [ConfigBlockKeyValue "provider" "\"prisma-client-js\""]`.
+--   Another example:
+--   ```
+--   datasource db {
+--     provider   = "postgresql"
+--     url        = env("DATABASE_URL")
+--     extensions = [hstore(schema: "myHstoreSchema"), pg_trgm, postgis(version: "2.1")]
+--   }
+--   ```
+--   The config block would be `ConfigBlock Datasource "db" [ ... ]`.
+data ConfigBlock = ConfigBlock ConfigBlockType Name [ConfigBlockKeyValue]
+  deriving (Show, Eq)
+
+configBlockName :: ConfigBlock -> Name
+configBlockName (ConfigBlock _ name _) = name
+
+configBlockKeyValues :: ConfigBlock -> [ConfigBlockKeyValue]
+configBlockKeyValues (ConfigBlock _ _ keyValues) = keyValues
```

Comment:
Since you are doing all this, I would probably just go with record instead:
```hs
data ConfigBlock = ConfigBlock {
  type :: ConfigBlockType,
  name :: Name,
  entries :: Entry
}
```